### PR TITLE
Feat 344 homapage should redirect to last newsflash

### PR DESCRIPTION
**Pull request by 9Syncopation**
added homepage initial-id received from store
and redirect >  ternary rendering

### CODE REVIEW

File: src/pages/HomePage.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ const HomePage: FC<IProps & RouteComponentProps<IRouteProps>> = ({ match }) => {
       <Box flexGrow={5} className={classes.widgetBox} position="relative">
         <OverlayLoader show={loading} />
         <FilterBar />
-        <WidgetsTemplate id={id} />
+        {id ? <WidgetsTemplate id={id} /> : <Redirect to={`/newsflash/${initialId}`} />}
```
Comment (yuvalbl) [inline src/pages/HomePage.tsx:45]: when using redirect in a component with view - please separate between rendered return to redirect. like:
```
if(something) {
  return <Redirect to={`/newsflash/${initialId}`} />
}

return (
  <div>Look at this amazing view!</div>
)
```

File: src/pages/HomePage.tsx [hunk 2]
```diff
@@ -30,10 +30,10 @@ const useStyles = makeStyles({
 
 const HomePage: FC<IProps & RouteComponentProps<IRouteProps>> = ({ match }) => {
   const classes = useStyles();
-  const id = match.params.id ? parseInt(match.params.id) : null;
   const store: RootStore = useStore();
+  const initialId = store?.newsFlashCollection[0]?.id;
```
Comment (yuvalbl) [inline src/pages/HomePage.tsx:34]: `newsFlashCollection` can be an empty array

Comment (9Syncopation) [reply src/pages/HomePage.tsx:34]: ok
is it a statement or a comment that need to be resolved ?

Comment (yuvalbl) [reply src/pages/HomePage.tsx:34]: Needs to be resolved, e.i. Make the code safe for all cases

File: src/pages/HomePage.tsx [hunk 3]
```diff
@@ -34,6 +34,10 @@ const HomePage: FC<IProps & RouteComponentProps<IRouteProps>> = ({ match }) => {
   const initialId = store?.newsFlashCollection[0]?.id;
   const id = match.params.id ? parseInt(match.params.id) : initialId;
```
Comment (yuvalbl) [inline src/pages/HomePage.tsx:35]: This is a logical mistake. if you'll set `id` to be `initialId` when param is not provided, it will never redirect!

Comment (yuvalbl) [reply src/pages/HomePage.tsx:35]: You can remove the `const id` and do:
```
  if (!match.params.id) {
    return <Redirect to={`/newsflash/${initialId}`} />;
  }
```

Comment (9Syncopation) [reply src/pages/HomePage.tsx:35]: The last suggested solution redirects to ...newsflsah/undefined
Due to the first time the app is loaded -initialId is undefined

I would like to discuss this issue  .

**Pull request merged**